cluster-autoscaler/utils.go

```diff
@@ -271,3 +271,15 @@ func fixNodeGroupSize(context *AutoscalingContext, currentTime time.Time) (bool,
 	}
 	return fixed, nil
 }
+
+// Enables cloud provider specific node protection for each node group.
+func protectNodes(context *AutoscalingContext) error {
+	for _, group := range context.CloudProvider.NodeGroups() {
```

Comment:
It seems easier to implement a no-op on the `NodeGroup` interface instead, so that there's no cloud provider specific logic in the main code path. It would also make it easier to add protection support on any cloud provider. But I can make this function aws specific if you think that's the right path to take.